File: ios/VllowDugout/Models/LeaderboardEntry.swift
import Foundation

nonisolated struct LeaderboardEntry: Identifiable, Hashable, Sendable {
    let id: UUID
    let rank: Int
    let playerName: String
    let avatarURL: String
    let vCoins: Int
    let drillsCompleted: Int
    let isCurrentUser: Bool

    init(
        id: UUID = UUID(),
        rank: Int,
        playerName: String,
        avatarURL: String = "",
        vCoins: Int,
        drillsCompleted: Int,
        isCurrentUser: Bool = false
    ) {
        self.id = id
        self.rank = rank
        self.playerName = playerName
        self.avatarURL = avatarURL
        self.vCoins = vCoins
        self.drillsCompleted = drillsCompleted
        self.isCurrentUser = isCurrentUser
    }
}

nonisolated enum LeaderboardPeriod: String, CaseIterable, Sendable {
    case weekly = "Weekly"
    case monthly = "Monthly"
    case allTime = "All-Time"
}

File: flutter_app/lib/models/leaderboard_entry.dart
import 'package:uuid/uuid.dart';

enum LeaderboardPeriod {
  weekly('Weekly'),
  monthly('Monthly'),
  allTime('All-Time');

  final String displayName;
  const LeaderboardPeriod(this.displayName);
}

class LeaderboardEntry {
  final String id;
  final int rank;
  final String playerName;
  final String avatarURL;
  final int vCoins;
  final int drillsCompleted;
  final bool isCurrentUser;

  LeaderboardEntry({
    String? id,
    required this.rank,
    required this.playerName,
    this.avatarURL = '',
    required this.vCoins,
    required this.drillsCompleted,
    this.isCurrentUser = false,
  }) : id = id ?? const Uuid().v4();

  @override
  bool operator ==(Object other) =>
      identical(this, other) || other is LeaderboardEntry && id == other.id;

  @override
  int get hashCode => id.hashCode;
}
